nixos/modules/services/networking/mosquitto.nix

```diff
@@ -170,23 +167,23 @@ let
             echo "$1:$(cat "$2")" >> "$file"
           }
         ''
-        + concatStringsSep "\n"
+        + lib.concatStringsSep "\n"
           (plainLines
-           ++ optional (plainLines != []) ''
+           ++ lib.optional (plainLines != []) ''
              ${cfg.package}/bin/mosquitto_passwd -U "$file"
            ''
            ++ hashedLines));
 
-  authPluginOptions = with types; submodule {
+  authPluginOptions = with lib.types; submodule {
```

Comment:
watch out, under this `with` there's no automatic checking.

Reply:
is your suggestion to get rid of this `with` statement as well?

Reply:
No, no suggestion as such.

I've found that when I make errors, it's in the following conditions:

1. `nix-instantiate --parse` succeeds
2. Because there's a symbol that was previously looked up in the top-level with
3. That now is "looked up" in the lower with
4. And there's no test of the NixOS module that would exercise the code.

`nixosTests.mosquitto` exists, so I'll definitely run that before merging. But the other "watch out" signals are there.